File: Chapter 05/5.6 Crypto Tracker/Before/PortfolioManager.cs
using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text.Json;

namespace CryptoTracker.Before
{
    public class PortfolioManager
    {
        public decimal CalculateTotalValue(decimal btcAmount)
        {
            // VIOLATION: Hard-coded dependency on a specific external API (CoinGecko)
            using var client = new HttpClient();
            client.DefaultRequestHeaders.Add("User-Agent", "C# App"); 

            // Synchronous .Result is bad practice, but typical for "hacked together" code
            var json = client.GetStringAsync("https://api.coingecko.com/api/v3/simple/price?ids=bitcoin&vs_currencies=usd").Result;
            
            // Logic is tangled with parsing specific external JSON format
            var priceData = JsonSerializer.Deserialize<Dictionary<string, Dictionary<string, decimal>>>(json);
            var currentPrice = priceData["bitcoin"]["usd"];
            
            return btcAmount * currentPrice;
        }
    }
}

File: Chapter 05/JS/section_5_6_crypto_tracker/before/portfolioManager.js
/**
 * THE CORE (Tightly Coupled).
 * This class is a major architectural liability because it mixes pure 
 * domain logic with network infrastructure concerns.
 */
class PortfolioManager {
    
    /**
     * Calculates the total USD value of a Bitcoin balance.
     * @param {number} btcAmount 
     * @returns {Promise<number>}
     */
    async calculateTotalValue(btcAmount) {
        // VIOLATION 1: Hard-coded infrastructure dependency.
        // Using native fetch means this code CANNOT run offline.
        const response = await fetch("https://api.coingecko.com/api/v3/simple/price?ids=bitcoin&vs_currencies=usd", {
            headers: { "User-Agent": "JS App" }
        });

        if (!response.ok) {
            throw new Error("Network response was not ok");
        }

        // VIOLATION 2: The logic is tangled with a specific external JSON format.
        const priceData = await response.json();
        const currentPrice = priceData.bitcoin.usd;
        
        return btcAmount * currentPrice;
    }
}

module.exports = PortfolioManager;